pyccel/ast/operators.py

```diff
@@ -1367,20 +1378,18 @@ def _calculate_dtype(cond, value_true, value_false):
         -------
         dtype : DataType
             The underlying datatype of the object.
-        precision : int
-            The precision of the datatype of the object.
         class_type : DataType
             The Python type of the object.
         """
-        if value_true.dtype in NativeNumeric and value_false.dtype in NativeNumeric:
-            dtype = max([value_true.dtype, value_false.dtype], key = NativeNumeric.index)
-        else:
-            dtype = value_true.dtype
+        if value_true.dtype is value_false.dtype and value_true.class_type is value_false.class_type:
+            return value_true.class_type
 
-        precision = max_precision([value_true, value_false])
+        try:
+            class_type = value_true.class_type + value_false.class_type
+        except NotImplementedError:
+            raise TypeError(f'Cannot determine the type of ({value_true}, {value_false})') #pylint: disable=raise-missing-from
 
-        class_type = value_true.class_type + value_false.class_type
-        return dtype, precision, class_type
+        return class_type
```

Comment:
As above: the docstring says that this function returns `(dtype, class_type)`.

Reply:
Fixed : ad383a79f